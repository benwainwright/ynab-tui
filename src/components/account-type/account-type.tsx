import { Text } from 'ink'
import { AccountType } from 'ynab'

export const AccountTypeLabel = ({ type }: { type: AccountType }) => {
  const types = {
    [AccountType.Checking]: 'Checking',
    [AccountType.AutoLoan]: 'Auto Loan',
    [AccountType.Cash]: 'Cash',
    [AccountType.CreditCard]: 'Credit Card',
    [AccountType.LineOfCredit]: 'Line of Credit',
    [AccountType.MedicalDebt]: 'Medical Debt',
    [AccountType.Mortgage]: 'Mortgage',
    [AccountType.OtherAsset]: 'Other Asset',
    [AccountType.OtherDebt]: 'Other Debt',
    [AccountType.OtherLiability]: 'Other Liability',
    [AccountType.PersonalLoan]: 'Personal Loan',
    [AccountType.Savings]: 'Savings',
    [AccountType.StudentLoan]: 'Student Loan',
  }
  return <Text>{types[type]}</Text>
}
