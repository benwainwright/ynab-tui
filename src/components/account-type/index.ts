export { AccountTypeLabel } from './account-type.tsx'
