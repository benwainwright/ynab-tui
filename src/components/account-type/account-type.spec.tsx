import { stripRender } from '@test-support'
import { AccountType } from 'ynab'
import { AccountTypeLabel } from './account-type.tsx'

describe('account type', () => {
  it.each`
    type                          | text
    ${AccountType.AutoLoan}       | ${'Auto Loan'}
    ${AccountType.Cash}           | ${'Cash'}
    ${AccountType.Checking}       | ${'Checking'}
    ${AccountType.CreditCard}     | ${'Credit Card'}
    ${AccountType.LineOfCredit}   | ${'Line of Credit'}
    ${AccountType.MedicalDebt}    | ${'Medical Debt'}
    ${AccountType.Mortgage}       | ${'Mortgage'}
    ${AccountType.OtherAsset}     | ${'Other Asset'}
    ${AccountType.OtherDebt}      | ${'Other Debt'}
    ${AccountType.OtherLiability} | ${'Other Liability'}
    ${AccountType.PersonalLoan}   | ${'Personal Loan'}
    ${AccountType.Savings}        | ${'Savings'}
    ${AccountType.StudentLoan}    | ${'Student Loan'}
    ${AccountType.PersonalLoan}   | ${'Personal Loan'}
    ${AccountType.Savings}        | ${'Savings'}
  `(
    'Should render the correct text for the given account type',
    ({ type, text }) => {
      const { lastFrame } = stripRender(<AccountTypeLabel type={type} />)

      expect(lastFrame()).toMatchSnapshot()
    },
  )
})
