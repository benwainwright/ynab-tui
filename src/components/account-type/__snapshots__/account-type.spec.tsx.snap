// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`account type > Should render the correct text for the given account type 1`] = `"Auto Loan"`;

exports[`account type > Should render the correct text for the given account type 2`] = `"Cash"`;

exports[`account type > Should render the correct text for the given account type 3`] = `"Checking"`;

exports[`account type > Should render the correct text for the given account type 4`] = `"Credit Card"`;

exports[`account type > Should render the correct text for the given account type 5`] = `"Line of Credit"`;

exports[`account type > Should render the correct text for the given account type 6`] = `"Medical Debt"`;

exports[`account type > Should render the correct text for the given account type 7`] = `"Mortgage"`;

exports[`account type > Should render the correct text for the given account type 8`] = `"Other Asset"`;

exports[`account type > Should render the correct text for the given account type 9`] = `"Other Debt"`;

exports[`account type > Should render the correct text for the given account type 10`] = `"Other Liability"`;

exports[`account type > Should render the correct text for the given account type 11`] = `"Personal Loan"`;

exports[`account type > Should render the correct text for the given account type 12`] = `"Savings"`;

exports[`account type > Should render the correct text for the given account type 13`] = `"Student Loan"`;

exports[`account type > Should render the correct text for the given account type 14`] = `"Personal Loan"`;

exports[`account type > Should render the correct text for the given account type 15`] = `"Savings"`;
